main/ahrs.go

```diff
@@ -0,0 +1,237 @@
+package main
+
+import "math"
+
+var sampleFreq float64 = 512.0
+var beta float64 = 0.1 //values of 0.02 or 0.025 have also been suggested
+var q0, q1, q2, q3 float64 = 1.0, 0.0, 0.0, 0.0
+var attitudeX, attitudeY, attitudeZ float32
+
+// Gets the current attitude represented as X (roll) and Y (pitch) values,
+// resulting in less computational load as the Z (yaw) value is not calculated.
+func GetCurrentAttitudeXY() (float32, float32) {
+	var q0a, q1a, q2a, q3a float64
+	q0a = q0
+	q1a = q1
+	q2a = q2
+	q3a = q3
+
+	attitudeX = float32(math.Atan2(2*(q0a*q1a+q2a*q3a), 1-2*((q1a*q1a)+(q2a*q2a))))
+	attitudeY = float32(math.Asin(2 * (q0a*q2a - q3a*q1a)))
+

```

Comment:
After removing the globals change

	attitudeX := float32(math.Atan2(2*(q0a*q1a+q2a*q3a), 1-2*((q1a*q1a)+(q2a*q2a))))
 	attitudeY := float32(math.Asin(2 * (q0a*q2a - q3a*q1a)))